tests/test_cli.py

```diff
@@ -46,3 +49,16 @@ def test_import_main():
     """
     with pytest.raises(SystemExit):
         import conda_package_handling.__main__  # noqa
+
+
+@pytest.mark.parametrize(
+    "artifact,n_files",
+    [("mock-2.0.0-py37_1000.conda", 43), ("mock-2.0.0-py37_1000.tar.bz2", 43)],
+)
+def test_list(artifact, n_files):
+    "Integration test to ensure `cph list` works correctly."
+    memfile = io.StringIO()
+    with redirect_stdout(memfile):
+        cli.main(["list", os.path.join(data_dir, artifact)])
+    memfile.seek(0)
+    assert n_files == sum(1 for line in memfile.readlines() if line.strip())
```

Comment:
why not use pytest's `capsys` here?

```suggestion
def test_list(artifact, n_files, capsys):
    "Integration test to ensure `cph list` works correctly."
    cli.main(["list", os.path.join(data_dir, artifact)])
    stdout, stderr = capsys.readouterr()
    assert n_files == sum(bool(line.strip()) for line in stdout.splitlines())
```